NDNc/pipeline/pipeline-interests-aimd.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * N-DISE: NDN for Data Intensive Science Experiments
+ * Author: Catalin Iordache <catalin.iordache@cern.ch>
+ *
+ * MIT License
+ *
+ * Copyright (c) 2021 California Institute of Technology
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <iostream>
+
+#include "pipeline-interests-aimd.hpp"
+
+namespace ndnc {
+PipelineInterestsAimd::PipelineInterestsAimd(Face &face, size_t size)
+    : PipelineInterests(face), m_maxSize{size},
+      m_lastDecrease{ndn::time::steady_clock::now()} {}
+
+PipelineInterestsAimd::~PipelineInterestsAimd() {
+    this->end();
+}
+
+void PipelineInterestsAimd::process() {
+    while (this->isValid()) {
+        face->loop();
+        onTimeout();
+
+        if (m_pit->size() >= m_maxSize) {
+            // std::cerr << "pipeline full \n";
+            continue;
+        }
+
+#ifdef DEBUG
+        assert(m_pit->size() <= m_maxSize);
+#endif
```

Comment:
I have two minor issues with this assert

1. `assert` is already disabled when `NDEBUG` is defined, I wonder why you're wrapping it in another `#ifdef` layer and don't use the standard `NDEBUG` macro instead?
2. the assertion comes immediately after an `if` that guarantees `m_pit->size()` will be strictly less than `m_maxSize`, so I'm not sure if the assertion adds much value here.

Reply:
I agree that this assertion is not necessary. I'll remove this assertion.